tests/integration/links/config_file/test_config_file_editing.py

```
@@ -103,3 +105,32 @@ def test_write_starter_creates_and_refuses_overwrite(tmp_path: Path) -> None:
+    """A basic string doubles every backslash; a literal one shows the pattern."""
+    rendered = render_metric({"pattern": r"#\s*noqa"})
+
+    assert rendered == "[[metrics]]\npattern = '#\\s*noqa'"
+
+
+@pytest.mark.parametrize(
+    "pattern",
+    (
+        pytest.param("it\\'s", id="a quote a literal string cannot hold"),
+        pytest.param("a\\b\nc", id="a newline a literal string cannot hold"),
+    ),
```

Ruff (0.16.1)

[warning] Wrong values type in `pytest.mark.parametrize` expected `list`

Use `list` for parameter values

(PT007)